pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -236,14 +244,11 @@ func (r *AtlasDatabaseUserReconciler) handleDeletion(
 		return true, workflow.OK()
 	}
 
-	_, err := atlasClient.DatabaseUsers.Delete(ctx, dbUser.Spec.DatabaseName, project.ID(), dbUser.Spec.Username)
-	if err != nil {
-		var apiError *mongodbatlas.ErrorResponse
-		if errors.As(err, &apiError) && apiError.ErrorCode != atlas.UsernameNotFound {
-			return true, workflow.Terminate(workflow.DatabaseUserNotDeletedInAtlas, err.Error())
-		}
-
+	err := dus.Delete(ctx, dbUser.Spec.DatabaseName, project.ID(), dbUser.Spec.Username)
+	if errors.Is(err, dbuser.ErrorNotFound) {
```

Comment:
we have zero unit test coverage in `atlasdatabaseuser_controller.go`. As we're touching pieces of code here, I'd like to see a unit test asserting we don't change behavior with this contribution.

Reply:
unit tests will come in next update

Reply:
Adding full reconciliation unit testing is too complex at this point. Instead I will just unit test `handleDeletion` here.